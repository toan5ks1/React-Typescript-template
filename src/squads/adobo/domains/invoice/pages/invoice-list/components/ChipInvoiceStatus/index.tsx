export * from "./ChipInvoiceStatus";
export { default } from "./ChipInvoiceStatus";
