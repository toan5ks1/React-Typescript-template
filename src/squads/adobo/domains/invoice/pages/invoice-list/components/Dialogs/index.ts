import DialogApprovePayment, {
    DialogApprovePaymentProps,
    FormApprovePaymentValues,
} from "./DialogApprovePayment";
import DialogCancelPayment, {
    DialogCancelPaymentProps,
    FormCancelPaymentValues,
} from "./DialogCancelPayment";
import DialogIssueInvoice, { DialogIssueInvoiceProps } from "./DialogIssueInvoice";
import DialogScheduledInvoiceHistory, {
    DialogScheduledInvoiceHistoryProps,
} from "./DialogScheduledInvoiceHistory";
import DialogVoidInvoice, {
    DialogVoidInvoiceProps,
    FormVoidInvoiceValues,
} from "./DialogVoidInvoice";

export type {
    DialogApprovePaymentProps,
    FormApprovePaymentValues,
    DialogCancelPaymentProps,
    FormCancelPaymentValues,
    DialogIssueInvoiceProps,
    DialogScheduledInvoiceHistoryProps,
    DialogVoidInvoiceProps,
    FormVoidInvoiceValues,
};
export {
    DialogApprovePayment,
    DialogCancelPayment,
    DialogIssueInvoice,
    DialogScheduledInvoiceHistory,
    DialogVoidInvoice,
};
