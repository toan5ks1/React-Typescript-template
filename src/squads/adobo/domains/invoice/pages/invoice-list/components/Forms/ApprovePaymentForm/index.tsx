export * from "./ApprovePaymentForm";
export { default } from "./ApprovePaymentForm";
