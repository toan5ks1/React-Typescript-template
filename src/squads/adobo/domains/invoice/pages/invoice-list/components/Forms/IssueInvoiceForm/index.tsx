export * from "./IssueInvoiceForm";
export { default } from "./IssueInvoiceForm";
