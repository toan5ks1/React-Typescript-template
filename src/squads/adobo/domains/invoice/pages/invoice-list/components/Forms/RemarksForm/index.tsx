export * from "./RemarksForm";
export { default } from "./RemarksForm";
