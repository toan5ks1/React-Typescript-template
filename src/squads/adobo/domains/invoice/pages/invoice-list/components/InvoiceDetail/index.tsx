export * from "./InvoiceDetail";
export { default } from "./InvoiceDetail";
