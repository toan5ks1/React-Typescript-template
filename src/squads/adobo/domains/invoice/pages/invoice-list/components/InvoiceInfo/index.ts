export * from "./InvoiceInfo";
export { default } from "./InvoiceInfo";
