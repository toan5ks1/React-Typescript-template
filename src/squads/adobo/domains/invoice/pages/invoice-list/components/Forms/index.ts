import ApprovePaymentForm from "./ApprovePaymentForm";
import IssueInvoiceForm from "./IssueInvoiceForm";
import RemarksForm from "./RemarksForm";

export { ApprovePaymentForm, IssueInvoiceForm, RemarksForm };
