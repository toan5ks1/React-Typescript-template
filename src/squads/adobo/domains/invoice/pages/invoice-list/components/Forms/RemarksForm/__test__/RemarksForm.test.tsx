import {
    TestHookFormProvider,
    TestQueryWrapper,
} from "src/squads/adobo/domains/invoice/test-utils/providers";

import { RemarksForm } from "src/squads/adobo/domains/invoice/pages/invoice-list/components/Forms";
import MuiPickersUtilsProvider from "src/squads/adobo/domains/invoice/providers/MuiPickersUtilsProvider";

import { render, screen } from "@testing-library/react";

jest.mock("src/squads/adobo/domains/invoice/hooks/useResourceTranslate", () => {
    return {
        __esModule: true,
        default: () => (translateKey: string) => translateKey,
    };
});

const renderComponent = () => {
    return render(
        <MuiPickersUtilsProvider>
            <TestQueryWrapper>
                <TestHookFormProvider>
                    <RemarksForm />
                </TestHookFormProvider>
            </TestQueryWrapper>
        </MuiPickersUtilsProvider>
    );
};

describe("<RemarksForm />", () => {
    beforeEach(() => {
        renderComponent();
    });

    it("should match snapshot", () => {
        expect(screen.queryByTestId("RemarksForm__form")).toMatchSnapshot();
    });

    it("should render remark input field", () => {
        expect(screen.getByTestId("RemarksForm__inputInvoiceRemark")).toBeInTheDocument();
    });
});
