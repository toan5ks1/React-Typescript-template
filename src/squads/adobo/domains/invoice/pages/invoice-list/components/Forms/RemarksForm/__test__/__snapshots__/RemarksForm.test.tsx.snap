// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RemarksForm /> should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-bottom: 8px;
}

.emotion-1 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-top: -8px;
  width: calc(100% + 8px);
  margin-left: -8px;
  -webkit-flex-basis: calc(100% + 8px);
  -ms-flex-preferred-size: calc(100% + 8px);
  flex-basis: calc(100% + 8px);
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: calc(100% + 8px);
  padding-bottom: 0px;
}

.emotion-1>.MuiGrid-item {
  padding-top: 8px;
}

.emotion-1>.MuiGrid-item {
  padding-left: 8px;
}

@media (min-width:600px) {
  .emotion-1 {
    -webkit-flex-basis: calc(100% + 8px);
    -ms-flex-preferred-size: calc(100% + 8px);
    flex-basis: calc(100% + 8px);
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: calc(100% + 8px);
  }
}

@media (min-width:900px) {
  .emotion-1 {
    -webkit-flex-basis: calc(100% + 8px);
    -ms-flex-preferred-size: calc(100% + 8px);
    flex-basis: calc(100% + 8px);
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: calc(100% + 8px);
  }
}

@media (min-width:1200px) {
  .emotion-1 {
    -webkit-flex-basis: calc(100% + 8px);
    -ms-flex-preferred-size: calc(100% + 8px);
    flex-basis: calc(100% + 8px);
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: calc(100% + 8px);
  }
}

@media (min-width:1536px) {
  .emotion-1 {
    -webkit-flex-basis: calc(100% + 8px);
    -ms-flex-preferred-size: calc(100% + 8px);
    flex-basis: calc(100% + 8px);
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: calc(100% + 8px);
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
  padding: 0px;
}

@media (min-width:600px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:900px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:1536px) {
  .emotion-2 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

.emotion-3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
}

.emotion-4 .TextFieldBase-readOnly .MuiInputBase-root {
  background-color: #fafafa;
}

.emotion-5 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(100% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, 9px) scale(1);
  -moz-transform: translate(14px, 9px) scale(1);
  -ms-transform: translate(14px, 9px) scale(1);
  transform: translate(14px, 9px) scale(1);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: none;
}

.emotion-5.Mui-focused {
  color: #1976d2;
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-5.Mui-error {
  color: #d32f2f;
}

.emotion-6 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 0 5px;
  padding-top: 1px;
  position: relative;
  border-radius: 4px;
  padding: 8.5px 14px;
}

.emotion-6.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-6:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-6:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-6.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-6.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-6.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-7 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 0;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: auto;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-top: 0;
  resize: none;
  padding: 0;
}

.emotion-7::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-7:focus {
  outline: 0;
}

.emotion-7:invalid {
  box-shadow: none;
}

.emotion-7::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-7:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-7.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-7:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-7:-webkit-autofill {
  border-radius: inherit;
}

.emotion-9 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-10 {
  float: unset;
  overflow: hidden;
  display: block;
  width: auto;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 0.01px;
  -webkit-transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: max-width 50ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  white-space: nowrap;
}

.emotion-10>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

<div
  class="MuiGrid-root MuiGrid-container emotion-0"
  data-testid="RemarksForm__form"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 emotion-1"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-2"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth emotion-3"
      >
        <div
          class="MuiFormControl-root MuiTextField-root  emotion-4"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary emotion-5"
            data-shrink="false"
            for="mui-1"
            id="mui-1-label"
          >
            remarks
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-multiline emotion-6"
          >
            <textarea
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline MuiInputBase-inputSizeSmall emotion-7"
              data-testid="RemarksForm__inputInvoiceRemark"
              id="mui-1"
              name="remarks"
              rows="6.474"
              style="height: 0px; overflow: hidden;"
            />
            <textarea
              aria-hidden="true"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputMultiline MuiInputBase-inputSizeSmall emotion-7"
              readonly=""
              style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding: 0px; width: 100%;"
              tabindex="-1"
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline emotion-9"
            >
              <legend
                class="emotion-10"
              >
                <span>
                  remarks
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
